client/blocks/domain-to-plan-nudge/index.jsx

```diff
@@ -0,0 +1,333 @@
+			hasFreePlan;      //has a free wpcom plan
+	}
+
+	getCartItem() {
+		const {
+			productSlug,
+			productId
+		} = this.props;
+
+		return {
+			product_id: productId,
+			product_slug: productSlug,
+			free_trial: false,
+			is_domain_registration: false
+		};
+	}
+
+	handleTransactionComplete( error, data ) {
+		const { siteId, translate, userCurrency, rawPrice, discountedRawPrice } = this.props;
+
+		const price = discountedRawPrice || rawPrice;
+
+		debug( 'transaction complete', error, data );
+		// see transaction-steps-mixin for matching checkout analytics
+		if ( error ) {
+			if ( this.state.noticeId ) {
+				this.props.removeNotice( this.state.noticeId );
+			}
+			const noticeAction = this.props.errorNotice(
+				translate( 'There was a problem completing the purchase. Please try again.' )
+			);
+			this.setState( {
+				isSubmitting: false,
+				noticeId: get( noticeAction, 'notice.noticeId' )
+			} );
+			this.props.recordTracksEvent(
+				'calypso_checkout_payment_error',
+				{ reason: error.message }
+			);
+			return;
+		}
+
+		const receiptId = data.receipt_id;
+
+		// ad tracking
+		const product = { ...this.getCartItem(), currency: userCurrency, cost: price, volume: 1 };
+		const cart = { products: [ product ], total_cost: price, currency: userCurrency };
+		recordOrder( cart, receiptId );
+
+		// tracks
+		this.props.recordTracksEvent( 'calypso_checkout_payment_success', {
+			coupon_code: '',
+			currency: userCurrency,
+			payment_method: storedCardPayment().paymentMethod,
+			total_cost: price
+		} );
+
+		this.props.recordTracksEvent( 'calypso_checkout_product_purchase', this.getCartItem() );
+
+		// redirect to thank you
+		page( `/checkout/thank-you/${ siteId }/${ receiptId }` );
+	}
+
+	oneClickUpgrade() {
+		const { siteId, storedCard, translate } = this.props;
+		const newPayment = storedCardPayment( storedCard );
+		const transaction = {
+			payment: newPayment
+		};
+
+		// set this to temporary so we don't clear the existing user cart on the server
+		let cart = emptyCart( siteId, { temporary: true } );
+
+		const personalCartItem = this.getCartItem();
+		cart = add( personalCartItem )( cart );
+
+		debug( 'purchasing with', cart, transaction );
+
+		if ( this.state.noticeId ) {
+			this.props.removeNotice( this.state.noticeId );
+		}
+		const noticeAction = this.props.infoNotice( translate( 'Submitting payment' ) );
+		this.setState( {
+			isSubmitting: true,
+			noticeId: get( noticeAction, 'notice.noticeId' )
+		} );
+
+		submitTransaction( { cart, transaction }, this.handleTransactionComplete );
+	}
+
+	render() {
+		if ( ! this.isVisible() ) {
+			return null;
+		}
+
+		const { isSubmitting } = this.state;
+
+		const {
+			discountedRawPrice,
+			productSlug,
+			rawDiscount,
+			rawPrice,
+			siteId,
+			storedCard,
+			translate,
+			userCurrency
+		} = this.props;
+
+		return (
+			<DismissibleCard
+				className="domain-to-plan-nudge"
+				preferenceName="domain-to-plan-nudge"
+				temporary
```

Comment:
This is a good question. It depends on how much we want to bug users. Maybe we could abtest it? Like if users have to close it on all pages to get rid of it, will there be higher churn? Or on the other hand, will it bring more one-click upgrades?

Also, if we plan to show this upgrade nudge only on a few pages, say 2-3, I think then it's okay to show this to users a bit more (ie have to dismiss every upgrade nudge separately) as they might miss what's written on it and close it subconsciously. However, if we plan to display it on each or a lot of pages, that would be pretty annoying so in that case, we should remove it from all on the first dismiss.

But anyways, we need to hear opinions from above as well. :)